assayist/processor/utils.py

```diff
@@ -84,7 +84,45 @@ def download_build(build_identifier, output_dir):
                 artifacts.append(file_path)
                 log.info(f'Downloaded {os.path.split(file_path)[-1]}')
 
-    return artifacts
+    return artifacts, build
+
+
+def download_source(build_info, output_dir):
+    """
+    Download the source (from dist-git) that was used in the specified build.
+
+    :param build_info: build information from koji.getBuild()
+    :param output_dir: the path to download the source to
+    """
+    # Make sure the git command is installed
+    _assert_command('git')
+
+    source_url = build_info.get('source')
+    if not source_url:
+        raise RuntimeError(f'Build {build_info["id"]} has no associated source URL.')
+
+    log.info(f'Downloading source for {build_info["id"]}')
+
+    url, _, commit_id = source_url.partition('#')
+    component = url.split('/')[-1]
+
+    cmd = ['git', 'clone', url]
+    process = subprocess.Popen(cmd, cwd=output_dir,
+                               stdout=subprocess.DEVNULL, stderr=subprocess.PIPE)
+
+    _, error_output = process.communicate()
+    error_output = error_output.decode('utf-8')
```

Comment:
BTW is there any reason why do you prefer `subprocess.Popen`
instead of shorter version with `subprocess.run`?

https://docs.python.org/3/library/subprocess.html#subprocess.run